OptimizelySDK/OptlyConfig/OptimizelyConfigService.cs

```diff
@@ -0,0 +1,225 @@
+﻿/* 
+ * Copyright 2019, Optimizely
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System.Collections.Generic;
+using System.Linq;
+using OptimizelySDK.Config;
+using OptimizelySDK.Entity;
+using OptimizelySDK.Utils;
+
+
+namespace OptimizelySDK.OptlyConfig
+{
+    public class OptimizelyConfigService
+    {
+        /// <summary>
+        /// Gets Map of all experiments except rollouts
+        /// </summary>
+        /// <param name="configObj">The project config</param>
+        /// <returns>Dictionary | Dictionary of experiment key and value as experiment object</returns>
+        private Dictionary<string, OptimizelyExperiment> GetExperimentsMap(ProjectConfig configObj)
+        {
+            var ExperimentsMap = new Dictionary<string, OptimizelyExperiment>();
+            var rolloutExperimentIds = GetRolloutExperimentIds(configObj.Rollouts);
+            var featureIdMap = GetVariableIdMap(configObj);
+
+            foreach (Experiment experiment in configObj.Experiments)
+            {
+                if (!rolloutExperimentIds.Contains(experiment.Id))
+                {
+                    var variationsMap = new Dictionary<string, OptimizelyVariation>();
+                    foreach (Variation variation in experiment.Variations)
+                    {
+                        var variablesMap = MergeFeatureVariables(configObj,
+                            featureIdMap,
+                            experiment.Id,
+                            variation.FeatureEnabled ?? false,
+                            variation.FeatureVariableUsageInstances);
+
+                        var optimizelyVariation = new OptimizelyVariation();
+                        optimizelyVariation.Id = variation.Id;
```

Comment:
Pass everything in `OptimizelyVariation` constructor and do assignment there. it's same for all.